### DISCUSSION TITLE
bash: docker: command not found

### DISCUSSION THREAD

**Discussion by Jolg42** [community/community#14542] [Codespaces]
Hi there!

I would like to experiment in GitHub Codespaces with Docker / docker-compose but I found that the commands are not found:
```
bash: docker: command not found
bash: docker-compose: command not found
```

Is there any workaround about this? Should I try to install Docker in Codespaces or is there an easier way?

I want to experiment with databases which are easy to setup with a docker-compose file, and call `docker-compose up -d`, then run tests against them.

Note: I tried using this project, which has a `.devcontainer` https://github.com/prisma/prisma/tree/main/.devcontainer
Maybe it's doable by tweaking the `.devcontainer`? Though it would mean I have to somehow reload the config every time I want to experiment?

**Comment by Jolg42** [comment]
I found out it just works if I start a Codespace on another project which doesn't have any `.devcontainer` config like https://github.com/prisma/prisma-engines

So I guess my config doesn't contain the setup to have docker and docker-compose, any help about how to solve that is welcome!

**Reply by bartr** [reply]
Your .devcontainer doesn't have Docker installed. 

The easiest way is to use the universal image
- in your .devcontainer/dockerfile
- change `typescript-node:0-${VARIANT}`
- to `universal`
- rebuild your Codespace and you should be good to go

More details are here: https://github.com/microsoft/vscode-dev-containers There are a lot of base images that you can customize to get exactly what you want/need. For just "playing around", the universal image is a good place to start.

Shameless plug on how we maintain our k3d in Codespaces images:  https://github.com/cse-labs/codespaces-images

**Comment by shayneburgess** [comment]
You can install docker in the container in the DockerFile.

You can also specify a docker-compose file instead of your DockerFile in the devcontainer.json (preferred)
